sterror/sterror.go

```diff
@@ -0,0 +1,157 @@
+package sterror
+
+import (
+	"errors"
+)
+
+type Op string
+
+func (o Op) String() string {
+	return string(o)
+}
+
+type Scope string
+
+func (s Scope) String() string {
+	return string(s)
+}
+
+type Info string
+
+func (i Info) String() string {
+	return string(i)
+}
+
+const (
+	colon     = ": "
+	hyphen    = " - "
+	separator = "\n"
+)
+
+const (
+	Undefined Scope = "Undefined"
+	Host      Scope = "Host"
+	Network   Scope = "Network"
+	Opts      Scope = "Opts"
+	Ospkg     Scope = "Ospkg"
+	Stlog     Scope = "Stlog"
+	Trust     Scope = "Trust"
+)
+
+/* Error-Wrapper
+- op: Function that raised the error
+- Scope: Package that raised the error
+- Info: additional Info passed as a string
+- Err: Wrapped underlying error.
+*/
+type Error struct {
+	Op    Op
+	Scope Scope
+	Info  Info
+	Err   error
+}
+
+// returns the error as a string.
```

Comment:
Doc comments of public functions begin with the name of the function
`Error implements the Error interface`